### PR TITLE
Pub 1001 check subscribers

### PR DESCRIPTION
**Pull request by kurt-uk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-1001

## API 
added new api that takes in an artefact to build a list of subscribers, returns a 202 accepted as long as the body is of type artefact.
the processing is done aync so as soon as the request comes in it will return 202 and open a new thread to complete the collect subscribers

collects all subscribers to a artefact then calls [Account management isAuthorised](https://github.com/hmcts/pip-account-management/blob/demo/src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountController.java)
which returns success bool if the person is able to see classified publication or not

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java [hunk 1]
```diff
@@ -96,4 +106,48 @@ UserSubscription collectSubscriptions(List<Subscription> subscriptions) {
...
+    }
+
+    private List<Subscription> extractSearchValue(Object caseObject) {
+        List<Subscription> subscriptionList = new ArrayList<>();
+        try {
+            subscriptionList.addAll(querySubscriptionValue(SearchType.CASE_ID.name(),
+                                                           ((LinkedHashMap) caseObject).get("caseNumber").toString()));
+            subscriptionList.addAll(querySubscriptionValue(SearchType.CASE_URN.name(),
+                                                           ((LinkedHashMap) caseObject).get("caseUrn").toString()));
+        } catch (NullPointerException ex) {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java:148]: Is there any way we can avoid catching the NullPointer here with explicit checks?

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by alberto-hortiguela — DISMISSED** [review]

**Review by alberto-hortiguela — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java (hunk 1 above)
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java:148]: this was added as case items dont seem to be required but we dont want to break on them, so NPE is caught to stop application breaking and the case item is skipped over

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Value("${service-to-service.account-management}")
+    private String url;
+
+    private static final String IS_AUTHORISED = "account/isAuthorised";
+
+    @Autowired
+    private WebClient webClient;
+
+    public Boolean isUserAuthenticated(String userId, ListType listType) {
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java:24]: i can indeed

**Review by alberto-hortiguela — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
looks good :)

**Pull request merged**